File: web_auto/automation_java/src/main/java/pages/FormPage.java
package pages;

import base.BasePage;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;

public class FormPage extends BasePage {

    @FindBy(id = "full-name")
    private WebElement fullNameInput;

    @FindBy(id = "department")
    private WebElement departmentSelect;

    @FindBy(id = "terms")
    private WebElement termsCheckbox;

    @FindBy(id = "role-student")
    private WebElement roleStudentRadio;

    @FindBy(id = "role-ta")
    private WebElement roleTaRadio;

    @FindBy(id = "submit-form")
    private WebElement submitButton;

    @FindBy(className = "success-message")
    private WebElement successMessage;

    public FormPage(WebDriver driver) {
        super(driver);
    }

    public void enterFullName(String name) {
        sendKeysToElement(fullNameInput, name);
    }

    public void selectDepartment(String value) {
        sendKeysToElement(departmentSelect, value); // Standard sendKeys works on <select> in some drivers, but standard Select is better.
    }

    public void clickTerms() {
        if (!termsCheckbox.isSelected()) {
            clickElement(termsCheckbox);
        }
    }

    public void selectRole(String role) {
        if (role.equalsIgnoreCase("ta")) {
            clickElement(roleTaRadio);
        } else {
            clickElement(roleStudentRadio);
        }
    }

    public void submitForm() {
        clickElement(submitButton);
    }

    public boolean isSuccessMessageDisplayed() {
        return isElementDisplayed(successMessage);
    }
}

File: web_auto/automation_js/pages/formPage.js
class FormPage {
  constructor(page) {
    this.page = page;
    this.fullNameInput = page.locator('#full-name');
    this.emailInput = page.locator('#email');
    this.ageInput = page.locator('#age');
    this.phoneInput = page.locator('#phone');
    this.departmentSelect = page.locator('#department');
    this.termsCheckbox = page.locator('#terms');
    this.submitButton = page.locator('button[type="submit"]');
    this.successMessage = page.locator('.success-message');
  }

  async enterFullName(name) {
    await this.fullNameInput.fill(name);
  }

  async enterEmail(email) {
    await this.emailInput.fill(email);
  }

  async enterAge(age) {
    await this.ageInput.fill(age.toString());
  }

  async enterPhone(phone) {
    await this.phoneInput.fill(phone);
  }

  async selectDepartment(value) {
    await this.departmentSelect.selectOption(value);
  }

  async clickTerms() {
    const isChecked = await this.termsCheckbox.isChecked();
    if (!isChecked) {
      await this.termsCheckbox.click();
    }
  }

  async submitForm() {
    await this.submitButton.click();
  }

  async waitForSuccessMessage() {
    await this.successMessage.waitFor({ state: 'visible', timeout: 5000 });
    return true;
  }
}

module.exports = FormPage;
